python/python/bystro/covariance/_covariance_np.py

```diff
@@ -106,3 +95,146 @@ def _fill_prior_options(self, prior_options: dict[str, Any]) -> dict[str, Any]:
             "iw_params": {"pnu": 2, "sigma": 1.0},
         }
         return {**default_options, **prior_options}
+
+
+class LinearShrinkageCovariance(BaseCovariance):
+
+    def fit(self, X: NDArray[np.float_]) -> "LinearShrinkageCovariance":
+        """
+        This fits a covariance matrix using the linear shrinkage approach 
+        to combine the empirical covariance matrix with a structured 
+        estimator.
+
+        Parameters
+        ----------
+        X : np.array-like, (n_samples, n_covariates)
```

Comment:
Let's add a note that we do not support missing values

Reply:
Where do you want to add it?

Reply:
I was just thinking right here: "The centered data. Must not contain any missing values (np.nan)"